main/src/ca/uwaterloo/flix/language/ast/TypedAst.scala

```diff
@@ -126,6 +128,8 @@ object TypedAst {
 
     case class LetRec(sym: Symbol.VarSym, ann: Annotations, mod: Ast.Modifiers, exp1: Expr, exp2: Expr, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
 
+    case class LocalDef(ann: Annotations, sym: Symbol.VarSym, fparams: List[FormalParam], declaredType: Type, declaredEff: Type, exp1: Expr, exp2: Expr, tpe: Type, eff: Type, loc: SourceLocation) extends Expr
```

Comment:
Do we need to keep the declared type/eff around? We have the real type and effect now.

Reply:
thats also related to my comment about the desugaring phase

Reply:
Yes, because `tpe` and `eff` is related to `exp2`, i.e., the type and effect of the whole expression, whereas the declared type is the result type of the local def (same with declared effect). Maybe they should be renamed to `arrowResTpe` and `arrowResEff`?

Reply:
Ah okay my bad, we have the in `exp1` of course